app/extensions/tus/flask_tus_cont.py

```diff
@@ -0,0 +1,383 @@
+            (key, value) = kv.split(' ')
+            metadata[key] = base64.b64decode(value).decode('utf-8')
+        return metadata
+
+    # Untested. Possibly unused.
+    def tus_proprietary_get_file_exists(self):
+        print('begin tus_proprietary_get_file_exists')
+        response = make_response('', 200)
+        if 'Upload-Metadata' in request.headers:
+            response.headers['Upload-Metadata'] = request.headers['Upload-Metadata']
+
+        metadata = self._parse_metadata()
+
+        if metadata.get('filename', None) is None:
+            return make_response('metadata filename is not set', 404)
+
+        (filename_name, extension) = os.path.splitext(metadata.get('filename'))
+        if filename_name.upper() in [
+            os.path.splitext(f)[0].upper()
+            for f in os.listdir(os.path.dirname(self.upload_folder))
+        ]:
+            response.headers['Tus-File-Name'] = metadata.get('filename')
+            response.headers['Tus-File-Exists'] = True
+        else:
+            response.headers['Tus-File-Exists'] = False
+        return response
+
+    def create_resource_id(self):
+        return str(uuid.uuid4())
+
+    def create_url(self, resource_id):
+        url_root = request.url_root
+        # assuming(!) we only ever want to upgrade from http to https, not the other way around
+        if (
+            'X-Forwarded-Proto' in request.headers
+            and request.headers.get('X-Forwarded-Proto') == 'https'
+            and url_root.lower().startswith('http:')
+        ):
+            url_root = 'https://' + url_root[7:]
+        return '{}/{}/{}'.format(url_root, self.upload_url, resource_id)
+
+    def tus_creation_1_create(self):
+        """Implements POST to create file according to Tus protocol Creation extension"""
+        print('begin tus_creation_1_create')
+        response = make_response('', 200)
+        response.headers['Tus-Resumable'] = self.tus_api_version
+        response.headers['Tus-Version'] = self.tus_api_version_supported
+
+        if 'Upload-Metadata' in request.headers:
+            response.headers['Upload-Metadata'] = request.headers['Upload-Metadata']
+
+        if request.headers.get('Tus-Resumable') is None:
+            self.app.logger.warning(
+                'Received File upload for unsupported file transfer protocol'
+            )
+            response.data = 'Received File upload for unsupported file transfer protocol'
+            response.status_code = 500
+            return response
+
+        # process upload metadata
+        metadata = self._parse_metadata()
+
+        if (
+            os.path.lexists(os.path.join(self.upload_folder, metadata.get('filename')))
+            and self.file_overwrite is False
+        ):
+            response.status_code = 409
+            return response
+
+        file_size = int(request.headers.get('Upload-Length', '0'))
+        resource_id = self.create_resource_id()
+
+        p = self.redis_connection.pipeline()
```

Comment:
Could p have a better name? I've never liked single char variable names

Reply:
agree with you conceptually here.... backstory on this code is it is a [clone of a library](https://pypi.org/project/Flask-Tus-Cont/) we needed to make a couple tweaks to.  so i am tempted to keep it the same.  (why?  i dunno... i doubt we are ever going to look to patch changes back in...).  counter-argument is that _linting_ has already moved things around in their code anyway!  haha. maybe i should drop a README in this dir explaining this history?

Reply:
Aha, if it's inherited and tweaked then yes, a README pointing to the source and documenting any changes sounds like a much better idea.

Reply:
i have added this readme